### PR TITLE
docs(blog): introduce Detox Copilot.

### PR DESCRIPTION
**Pull request by asafkorem**

### CODE REVIEW

File: website/blog/2024-09-30-detox-copilot-is-out.md [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+
+# Introducing Detox Copilot: Natural Language Testing for Detox
+
+We're excited to announce the release of **Detox Copilot**, a groundbreaking feature that brings natural language testing to Detox. With Detox Copilot, you can now write end-to-end tests using plain English commands, making test creation more intuitive and accessible than ever before.
+
+## Revolutionizing Testing with Natural Language
+
+Detox Copilot leverages advanced Large Language Models (LLMs) to interpret natural language instructions and translate them into Detox actions and assertions. This means you can describe your test scenarios in everyday language, aligning perfectly with **Behavior-Driven Development (BDD)** principles.
+
+### Why Natural Language Testing?
```
Comment (d4vidi) [inline website/blog/2024-09-30-detox-copilot-is-out.md:15]: I think we also need to preset the benefits of the reduced maintenance cost (i.e. thanks to the decoupling from specific test ID's, not to mention xpath, etc). Let's try to come up with some examples? It's too important to be overlooked 🤔

File: website/blog/2024-09-30-detox-copilot-is-out.md [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+- **Faster Test Creation**: Reduce the time spent writing and maintaining complex test scripts.
+- **Enhanced Test Coverage**: Encourage more comprehensive testing by lowering the barrier to writing tests.
+
+![Detox Copilot in action GIF](/img/blog/copilot-demo.gif)
+
+## Key Features of Detox Copilot
+
+### Write Tests in Plain English
+
+Detox Copilot allows you to write tests using natural language commands. Each step corresponds to a specific action or assertion within your app.
```
Comment (d4vidi) [inline website/blog/2024-09-30-detox-copilot-is-out.md:27]: good stuff!

File: website/blog/2024-09-30-detox-copilot-is-out.md [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+      'Tap on the "Register" button',
+      'Verify that the "Welcome, Jane!" message is displayed'
+    );
+  });
+});
+```
+
+### Seamless Integration with Detox
+
+Detox Copilot is built into Detox and requires no additional installation. Simply initialize it in your test setup, and you're ready to start writing natural language tests.
```
Comment (d4vidi) [inline website/blog/2024-09-30-detox-copilot-is-out.md:49]: good stuff!

File: website/blog/2024-09-30-detox-copilot-is-out.md [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+
+While Detox Copilot uses LLMs to interpret instructions, it is designed to be LLM-agnostic. This means you can connect it to your preferred language model service, offering flexibility and future-proofing your testing strategy.
+
+## Getting Started with Detox Copilot
+
+### Step 1: Initialize Detox Copilot
+
+Initialize Detox Copilot in your test setup by implementing a `PromptHandler` that connects to your LLM service. This abstraction allows Detox Copilot to communicate with any LLM you choose, promoting versatility across different testing frameworks.
+
+In our [Testing with Copilot guide](https://wix.github.io/Detox/docs/next/guide/testing-with-copilot), you can find implementation examples and more detailed descriptions for this part.
```
Comment (d4vidi) [inline website/blog/2024-09-30-detox-copilot-is-out.md:61]: `/next/...`?

File: website/blog/2024-09-30-detox-copilot-is-out.md [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+
+**Returning Values Example:**
+
+```javascript
+it('should display the correct page title', async () => {
+  // Use Copilot to navigate and retrieve the page title
+  const pageTitle = await copilot.perform(
+    'Launch the app',
+    'Navigate to the "Profile" page',
+    'Return the text of the page title'
```
Comment (d4vidi) [inline website/blog/2024-09-30-detox-copilot-is-out.md:119]: Woah, can't believe this actually works (!)

File: website/blog/2024-09-30-detox-copilot-is-out.md [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+  );
+
+  // Use the returned value in a hard-coded assertion
+  jestExpect(pageTitle).toBe('Profile');
```
Comment (d4vidi) [inline website/blog/2024-09-30-detox-copilot-is-out.md:123]: Hmm isn't this discouraged? I wonder

File: website/blog/2024-09-30-detox-copilot-is-out.md [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+});
+```
+
+In the example above, Copilot executes the steps and returns the text of the page title, which you can then use in your own assertions.
+
+You can also use Copilot for locating the elements and then use them in your hard-coded assertions.
+
+**Locating Elements Example:**
```
Comment (d4vidi) [inline website/blog/2024-09-30-detox-copilot-is-out.md:131]: This is really cool!

File: website/blog/2024-09-30-detox-copilot-is-out.md (hunk 6 above)
Comment (asafkorem) [reply website/blog/2024-09-30-detox-copilot-is-out.md:123]: hmm, yeahhh.. you're right. not the best example for this ability. I'll try to come with a better one

File: website/blog/2024-09-30-detox-copilot-is-out.md (hunk 1 above)
Comment (asafkorem) [reply website/blog/2024-09-30-detox-copilot-is-out.md:15]: 100%

File: website/blog/2024-09-30-detox-copilot-is-out.md (hunk 4 above)
Comment (asafkorem) [reply website/blog/2024-09-30-detox-copilot-is-out.md:61]: yeah, `next/` is the most up-to-date docs (on master). I'll try to use the latest published version instead.

File: website/blog/2024-09-30-detox-copilot-is-out.md [hunk 8]
```diff
@@ -0,0 +1,219 @@
...
+
+## Getting Started with Detox Copilot
+
+### Step 1: Initialize Detox Copilot
+
+Initialize Detox Copilot in your test setup by implementing a `PromptHandler` that connects to your LLM service. This abstraction allows Detox Copilot to communicate with any LLM you choose, promoting versatility across different testing frameworks.
+
+In our [Testing with Copilot guide](https://wix.github.io/Detox/docs/next/guide/testing-with-copilot), you can find implementation examples and more detailed descriptions for this part.
+
+### Step 2: Write Tests Using Natural Language
```
Comment (d4vidi) [inline website/blog/2024-09-30-detox-copilot-is-out.md:63]: Shouldn't this be a part of the "Testing with CoPilot" guide?

File: website/blog/2024-09-30-detox-copilot-is-out.md [hunk 9]
```diff
@@ -0,0 +1,219 @@
...
+  const cartBadge = element(by.id('cart-badge'));
+  await expect(cartBadge).toHaveText('1');
+  
+  // Continue with Copilot
+  await copilot.perform(
+    'Navigate to the "Cart" page',
+    'Verify that the product is listed in the cart'
+  );
+});
+```
```
Comment (d4vidi) [inline website/blog/2024-09-30-detox-copilot-is-out.md:106]: BTW, what about attributes inspection (as prompts)?
- Color assertions
- Element size, opacity

Maybe this is a 1st in a whole new set of questions we can ask, IDK

**Comment by d4vidi** [comment]
+ Need'a update the dir-structure doc
image

**Comment by d4vidi** [comment]
Not _directly_ related of course, but I think we REALLY need to offer the community with out-of-the-box implementations for this... If the goal was to call for contributions on that, then let's voice that more explicitly 💬

image

File: website/blog/2024-09-30-detox-copilot-is-out.md (hunk 9 above)
Comment (asafkorem) [reply website/blog/2024-09-30-detox-copilot-is-out.md:106]: That was the example for underlying getAttributes - https://github.com/wix/Detox/pull/4596/files#r1781100119

Comment (asafkorem) [reply website/blog/2024-09-30-detox-copilot-is-out.md:106]: but, sounds good as an alternative for the previous one

File: website/blog/2024-09-30-detox-copilot-is-out.md (hunk 8 above)
Comment (asafkorem) [reply website/blog/2024-09-30-detox-copilot-is-out.md:63]: It should be part of both IMO 😄

Comment (asafkorem) [reply website/blog/2024-09-30-detox-copilot-is-out.md:63]: but I agree that the long list of examples should be maintained on the guide. I'll keep some good one on the blog and the rest will be in the guide

File: website/blog/2024-09-30-detox-copilot-is-out.md (hunk 4 above)
Comment (asafkorem) [reply website/blog/2024-09-30-detox-copilot-is-out.md:61]: replaced with the current version docs

**Comment by asafkorem** [comment]
> Not _directly_ related of course, but I think we REALLY need to offer the community with out-of-the-box implementations for this... If the goal was to call for contributions on that, then let's voice that more explicitly 💬
> 
> image

Good idea. I'll add a few words to the blogpost, and a few potentially-popular handlers (OpenAI & Anthropic) for `detox-copilot` repo as a start. extremely easy to implement.
